### PR TITLE
Bump charm library version in extensions

### PR DESCRIPTION
**Pull request by weiiwang01**
Bump the version of the `loki_k8s.loki_push_api` charm libraries in `django-framework`, `fastapi-framework`, `flask-framework`, and `go-framework` to `v1`. Also, bump the `paas-app-charmer` library version to `>=1.1` in the corresponding init templates.

This upgrade introduces several new features in `paas-app-charmer`, including the Pebble log forwarding feature and various bug fixes.

### CODE REVIEW

**Review by jdkandersson — APPROVED** [review]

File: charmcraft/templates/init-django-framework/requirements.txt.j2
Comment (jdkandersson) [inline charmcraft/templates/init-django-framework/requirements.txt.j2:1]: The Charmcraft team gave feedback that we should include a link to the integration tests that verify backwards compatibility

File: charmcraft/templates/init-django-framework/requirements.txt.j2 [hunk 1]
```diff
@@ -1 +1 @@
-paas-app-charmer==1.*
+paas-app-charmer>1.1,<2
```
Comment (lengau) [inline charmcraft/templates/init-django-framework/requirements.txt.j2:1]:
```suggestion
paas-app-charmer>=1.1,<2
```

And similar for the others

**Review by lengau — CHANGES_REQUESTED** [review]

File: charmcraft/templates/init-django-framework/requirements.txt.j2 (hunk 1 above)
Comment (weiiwang01) [reply charmcraft/templates/init-django-framework/requirements.txt.j2]: Oops, sorry, updated, thanks!

**Review by jdkandersson — APPROVED** [review]

**Review by tigarmo — APPROVED** [review]

File: charmcraft/extensions/app.py [hunk 2]
```diff
@@ -31,7 +31,7 @@ class _AppBase(Extension):
         {"lib": "traefik_k8s.ingress", "version": "2"},
         {"lib": "observability_libs.juju_topology", "version": "0"},
         {"lib": "grafana_k8s.grafana_dashboard", "version": "0"},
-        {"lib": "loki_k8s.loki_push_api", "version": "0"},
+        {"lib": "loki_k8s.loki_push_api", "version": "1"},
```
Comment (lengau) [inline charmcraft/extensions/app.py:34]: Please address the test issues this raised :-)

**Review by lengau — APPROVED** [review]

File: charmcraft/extensions/app.py (hunk 2 above)
Comment (jdkandersson) [reply charmcraft/extensions/app.py:34]: @lengau it looks like the Python 3.12 tests weren't even run?

File: charmcraft/templates/init-django-framework/requirements.txt.j2
Comment (weiiwang01) [reply charmcraft/templates/init-django-framework/requirements.txt.j2:1]: Here's a link to an integration test in `paas-app-charmer` that checks `v0` exists, `v1` exists, and both `v0` and `v1` exist on Juju 3.3-3.5. Is this sufficient?

https://github.com/canonical/paas-app-charmer/actions/runs/10851499274

File: charmcraft/extensions/app.py (hunk 2 above)
Comment (weiiwang01) [reply charmcraft/extensions/app.py:34]: I have fixed all the issues with tests, thanks!

Comment (lengau) [reply charmcraft/extensions/app.py:34]: @jdkandersson the setup is a bit weird right now, as we use tox and skip environments inside it, but across all the Linux versions it runs all the Python tests. We're moving towards a base testing system that'll be easier to follow across all of our repos.

File: charmcraft/templates/init-django-framework/requirements.txt.j2
Comment (lengau) [reply charmcraft/templates/init-django-framework/requirements.txt.j2:1]: Thanks!

**Pull request merged**